src/shared/bus-unit-util.c

```diff
@@ -1315,6 +1315,30 @@ static int bus_append_capabilities(sd_bus_message *m, const char *field, const c
         return 1;
 }
 
+static int bus_append_cpu_affinity(sd_bus_message *m, const char *field, const char *eq) {
+        _cleanup_(cpu_set_done) CPUSet cpuset = {};
+        _cleanup_free_ uint8_t *array = NULL;
+        size_t allocated;
+        int r;
+
+        if (eq && streq(eq, "numa")) {
+                r = sd_bus_message_append(m, "(sv)", "CPUAffinityFromNUMA", "b", true);
+                if (r < 0)
+                        return bus_log_create_error(r);
+                return r;
+        }
+
+        r = parse_cpu_set(eq, &cpuset);
+        if (r < 0)
+                return log_error_errno(r, "Failed to parse %s value: %s", field, eq);
+
+        r = cpu_set_to_dbus(&cpuset, &array, &allocated);
+        if (r < 0)
+                return log_error_errno(r, "Failed to serialize %s: %m", field);
+
+        return bus_append_byte_array(m, field, array, allocated);
```

Comment:
`bus_append_parse_cpu_set()` can be used ?

Reply:
Thanks, that's a nice simplification.